### PR TITLE
preserve all

### PR DESCRIPTION
**Pull request by kolia**
When adding default Revise and PProf (which are totally arbitrary things to put into the base project environment...), add `preserve=Pkg.PRESERVE_ALL` kwarg per @ssfrr's suggestion.

### CODE REVIEW

**Comment by kleinschmidt** [comment]
`sysimage.jl` also does Pkg operations, is there a chance that it would also modify the manifest in the process?

**Comment by kolia** [comment]
> `sysimage.jl` also does Pkg operations, is there a chance that it would also modify the manifest in the process?

`sysimage.jl` just calls `create_sysimage`; I assume that doesn't mess with versions but I could be wrong.

There are a couple of calls to `Pkg.rm` on lines 19 and 23 of `Dockerfile.template`, but inferring from the Pkg.jl docs seems like removing dependencies shouldn't change versions either.

So it seems likely to be just the adding of Revise and friends that can mess with versions, but I haven't done any experiments to check.

**Comment by kleinschmidt** [comment]
`sysimage.jl` calls `Pkg.add` here: https://github.com/beacon-biosignals/julia_pod/blob/ba16afc42d99881d3689ed20c83f0b22bb816b3f/add_me_to_your_PATH/sysimage.jl#L22

**Comment by kleinschmidt** [comment]
and it's run in the project environment at least if I'm understanding correctly: https://github.com/beacon-biosignals/julia_pod/blob/ba16afc42d99881d3689ed20c83f0b22bb816b3f/add_me_to_your_PATH/Dockerfile.template#L74-L75

**Comment by ssfrr** [comment]
yeah, agreed it should also be added to that line in sysimage.jl. It [is indeed called in the project context](https://github.com/beacon-biosignals/julia_pod/blob/ba16afc42d99881d3689ed20c83f0b22bb816b3f/add_me_to_your_PATH/Dockerfile.template#L75)

Otherwise LGTM.

File: add_me_to_your_PATH/Dockerfile.template [hunk 1]
```diff
@@ -93,7 +93,7 @@ FROM sysimage-image as precompile-image
 ARG ADD_UTILS="true"
 RUN --mount=type=secret,id=github_token \
     if [ "$ADD_UTILS" = "true" ]; then \
-        julia -e 'using Pkg; Pkg.add("Revise"); Pkg.add("PProf"); Pkg.instantiate()'; \
+        julia -e 'using Pkg; Pkg.add("Revise"; preserve=Pkg.PRESERVE_ALL); Pkg.add("PProf"; preserve=Pkg.PRESERVE_ALL); Pkg.instantiate()'; \
```
Comment (ssfrr) [inline add_me_to_your_PATH/Dockerfile.template:96]: what's the logic of doing the `Pkg.add` before `Pkg.instantiate`? I would think you'd first install all the versions from the manifest, and _then_ add the additional packages (while maintaining the manifest versions). Is it possible here that the `Pkg.add` will install the latest versions of the packages and then `Pkg.instantiate `will lock them to the manifest afterwards?

Comment (kolia) [reply add_me_to_your_PATH/Dockerfile.template:96]: Looking at the `Pkg.jl` docs, looks like `instantiate` does not `resolve` anything if a Manifest already exists, which it should after calling `Pkg.add`.

My guess is that that `Pkg.instantiate` call is a no-op here, but I haven't checked to be sure.

**Comment by kolia** [comment]
In practice as soon as a `julia-1.9` docker image is made available, we'll probably want to switch to that and gut the whole sysimage part out of julia_pod, which will simplify things a lot. In the mean time...

**Review by ssfrr — APPROVED** [review]
lgtm

**Pull request merged**